integration/test-integration/src/test/java/org/wso2/choreo/connect/tests/testcases/standalone/router/EnvoyHttpFilterTestCase.java

```diff
@@ -51,7 +51,7 @@ public void checkHeadersSentToBackend() throws Exception {
 
         // Request headers received by the backend
         JSONObject headersSentToBackend = new JSONObject(response.getData());
-        Assert.assertEquals(headersSentToBackend.length(), 8, "Unexpected number of headers received by the backend");
+        Assert.assertEquals(headersSentToBackend.length(), 9, "Unexpected number of headers received by the backend");
```

Comment:
Why do we increment this?
If the rate limit headers are the reason, we should not let them pass to the backend